confmap/confmap_test.go

```diff
@@ -1151,3 +1173,160 @@ func TestMapMerge(t *testing.T) {
 		})
 	}
 }
+
+func TestConfIsNil(t *testing.T) {
+	testCases := []struct {
+		name         string
+		input        map[string]any
+		expectIsNil  bool
+		subExpectNil bool
+		subExpectErr string
+	}{
+		{
+			name:         "nil input",
+			input:        nil,
+			expectIsNil:  true,
+			subExpectNil: true,
+		},
+		{
+			name:         "empty map",
+			input:        map[string]any{},
+			expectIsNil:  false,
+			subExpectNil: true,
+		},
+		{
+			name:         "nil subkey",
+			input:        map[string]any{"foo": nil},
+			expectIsNil:  false,
+			subExpectNil: true,
+		},
+		{
+			name:         "empty subkey",
+			input:        map[string]any{"foo": map[string]any{}},
+			expectIsNil:  false,
+			subExpectNil: false,
+		},
+		{
+			name:         "non-empty map",
+			input:        map[string]any{"foo": map[string]any{"bar": 42}},
+			expectIsNil:  false,
+			subExpectNil: false,
+		},
+		{
+			name:         "non-map subkey",
+			input:        map[string]any{"foo": 123},
+			expectIsNil:  false,
+			subExpectErr: "unexpected sub-config value kind",
+		},
+	}
+
+	const subKey = "foo"
```

Comment:
Nit: we only use this constant once below, we should use it in the maps above or just use `"foo"` everywhere.